Unable to Run tests with non-standard maven settings

**Describe the bug**
When using non-standard maven settings, typically the user maven repository is no configured at `$HOME/.m2`, the test fails with exception unable to find the project artifact at $HOME/.m2/repository

**Expected behavior**
Test should run successfully

**Actual behavior**
Test fails with the following exception 

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.acme.quarkus.sample.HelloResourceTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.499 s <<< FAILURE! - in org.acme.quarkus.sample.HelloResourceTest
[ERROR] testHelloEndpoint  Time elapsed: 0.009 s  <<< ERROR!
org.junit.jupiter.api.extension.TestInstantiationException: TestInstanceFactory [io.quarkus.test.junit.QuarkusTestExtension] failed to instantiate test class [org.acme.quarkus.sample.HelloResourceTest]: Failed to create the boostrap class loader
Caused by: java.lang.IllegalStateException: Failed to create the boostrap class loader
Caused by: io.quarkus.bootstrap.BootstrapException: Failed to create the deployment classloader for org.acme.quarkus.sample:my-quarkus-project::jar:1.0-SNAPSHOT
Caused by: io.quarkus.bootstrap.resolver.AppModelResolverException: Failed to read descriptor of org.acme.quarkus.sample:my-quarkus-project:jar:1.0-SNAPSHOT
Caused by: org.eclipse.aether.resolution.ArtifactDescriptorException: Failed to read artifact descriptor for org.acme.quarkus.sample:my-quarkus-project:jar:1.0-SNAPSHOT
Caused by: org.apache.maven.model.resolution.UnresolvableModelException: Could not transfer artifact io.quarkus:quarkus-bom:pom:0.21.1 from/to central (https://repo.maven.apache.org/maven2): /.m2/repository/io/quarkus/quarkus-bom/0.21.1/quarkus-bom-0.21.1.pom.part.lock (No such file or directory)
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not transfer artifact io.quarkus:quarkus-bom:pom:0.21.1 from/to central (https://repo.maven.apache.org/maven2): /.m2/repository/io/quarkus/quarkus-bom/0.21.1/quarkus-bom-0.21.1.pom.part.lock (No such file or directory)
Caused by: org.eclipse.aether.transfer.ArtifactTransferException: Could not transfer artifact io.quarkus:quarkus-bom:pom:0.21.1 from/to central (https://repo.maven.apache.org/maven2): /.m2/repository/io/quarkus/quarkus-bom/0.21.1/quarkus-bom-0.21.1.pom.part.lock (No such file or directory)
Caused by: java.io.FileNotFoundException: /.m2/repository/io/quarkus/quarkus-bom/0.21.1/quarkus-bom-0.21.1.pom.part.lock (No such file or directory)
```


**To Reproduce**
Steps to reproduce the behavior:
1.  Delete the maven settings at ~/.m2/settings.xml
2. Configure global maven settings  $MAVEN_HOME/settings.xml to have `localRepository` configured at anyother place than $HOME/.m2/repository
3. Create or take any Quarkus quickstart projects and run the `mvn clean test`

**Configuration**
Maven configuration as mentioned above

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: all supported OS
 - Output of `java -version`: 1.8(openjdk)
 - GraalVM version (if different from Java):  19.1.1
 - Quarkus version or git rev: 0.21.1

--- a/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/MavenRepoInitializer.java
+++ b/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/MavenRepoInitializer.java
@@ -77,13 +77,13 @@ public class MavenRepoInitializer {
     private static final String BASEDIR = "basedir";
     private static final String MAVEN_CMD_LINE_ARGS = "MAVEN_CMD_LINE_ARGS";
     private static final String DOT_M2 = ".m2";
-    private static final String MAVEN_HOME = "maven.home";
-    private static final String M2_HOME = "M2_HOME";
+    private static final String MAVEN_DOT_HOME = "maven.home";
+    private static final String MAVEN_HOME = "MAVEN_HOME";
     private static final String SETTINGS_XML = "settings.xml";
 
     private static final String userHome = PropertyUtils.getUserHome();
     private static final File userMavenConfigurationHome = new File(userHome, DOT_M2);
-    private static final String envM2Home = System.getenv(M2_HOME);
+    private static final String envM2Home = System.getenv(MAVEN_HOME);
     private static final File USER_SETTINGS_FILE;
     private static final File GLOBAL_SETTINGS_FILE;
 
@@ -109,7 +109,7 @@ public class MavenRepoInitializer {
         File f = userSettings != null ? resolveUserSettings(userSettings) : new File(userMavenConfigurationHome, SETTINGS_XML);
         USER_SETTINGS_FILE = f != null && f.exists() ? f : null;
 
-        f = globalSettings != null ? resolveUserSettings(globalSettings) : new File(PropertyUtils.getProperty(MAVEN_HOME, envM2Home != null ? envM2Home : ""), "conf/settings.xml");
+        f = globalSettings != null ? resolveUserSettings(globalSettings) : new File(PropertyUtils.getProperty(MAVEN_DOT_HOME, envM2Home != null ? envM2Home : ""), "conf/settings.xml");
         GLOBAL_SETTINGS_FILE = f != null && f.exists() ? f : null;
     }
 

